### ISSUE TITLE
Failed import of useReduceMotion with @rails/webpacker

### ISSUE THREAD

**Issue by dsvanlani** [mantinedev/mantine#192]
Issue in @mantine/hooks, v2.2.0, on Mac os Catalina, Firefox Developer.

I'm using @rails/webpacker and upgraded mantine version 1.0.0 to 2.2.0.  I got this error from a previously compiling file.

```
ERROR in ./app/client/src/javascript/login/Signup/MarketingOverview.tsx 5:21-37
"export 'useReducedMotion' was not found in '@mantine/hooks'
    at HarmonyImportSpecifierDependency._getErrors (/Users/dsvanlani/src/bindplane/services/front-end/node_modules/webpack/lib/dependencies/HarmonyImportSpecifierDependency.js:109:11)
    at HarmonyImportSpecifierDependency.getErrors (/Users/dsvanlani/src/bindplane/services/front-end/node_modules/webpack/lib/dependencies/HarmonyImportSpecifierDependency.js:68:16)
    at Compilation.reportDependencyErrorsAndWarnings (/Users/dsvanlani/src/bindplane/services/front-end/node_modules/webpack/lib/Compilation.js:1463:22)
    at hooks.finishModules.callAsync.err (/Users/dsvanlani/src/bindplane/services/front-end/node_modules/webpack/lib/Compilation.js:1258:10)
    at AsyncSeriesHook.eval [as callAsync] (eval at create (/Users/dsvanlani/src/bindplane/services/front-end/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:15:1)
    at AsyncSeriesHook.lazyCompileHook (/Users/dsvanlani/src/bindplane/services/front-end/node_modules/tapable/lib/Hook.js:154:20)
    at Compilation.finish (/Users/dsvanlani/src/bindplane/services/front-end/node_modules/webpack/lib/Compilation.js:1253:28)
    at hooks.make.callAsync.err (/Users/dsvanlani/src/bindplane/services/front-end/node_modules/webpack/lib/Compiler.js:672:17)
    at _done (eval at create (/Users/dsvanlani/src/bindplane/services/front-end/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:9:1)
    at _err0 (eval at create (/Users/dsvanlani/src/bindplane/services/front-end/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:20:22)
    at _addModuleChain (/Users/dsvanlani/src/bindplane/services/front-end/node_modules/webpack/lib/Compilation.js:1185:12)
    at processModuleDependencies.err (/Users/dsvanlani/src/bindplane/services/front-end/node_modules/webpack/lib/Compilation.js:1097:9)
    at process._tickCallback (internal/process/next_tick.js:61:11)
```

I since made sure I was on the most recent version of @rails/webpacker (5.4.0).  

This import was working fine before upgrading and I'm not sure what would be the issue.

My import:
```
import { useReducedMotion } from '@mantine/hooks';
```

and in use:

```
 const reduceMotion = useReducedMotion();
```

**Comment by rtivital** [comment]
Hi David, since version 1.0.0 we've changed a build step and now the primary way to use Mantine is via cjs modules, this may cause the issue with your build system. Can you try out a few things:

- import all hooks and logging them into console – `import * as hooks from '@mantine/hooks'; console.log(hooks);` – is useReducedMotion hook there?
- use @mantine/core component which depends on this hook and see if it works – for example, Burger – https://mantine.dev/core/burger/

**Comment by dsvanlani** [comment]
Hi Vitaly - thanks for getting back to me.

I tried your suggestions and uncovered a couple things.

I can use the import statement (`import * as hooks from '@mantine/hooks'`) and webpacker compiles successfully, but when I go to use the import (`console.log(hooks)`) it makes the whole component unavailable.  That is, it does not render in my app.

Additionally everywhere in my app where I'm using any mantine component or hook, that component is not found and does not render.

Anyway I'm still investigating if there are changes I can make to my configuration files.  Do you have any other suggestions?

**Comment by rtivital** [comment]
> That is, it does not render in my app

Is there any output in browser console for `console.log(hooks)`?

I do not have any experience with rails, so maybe you can create a new empty project with your rails/webpack configuration so that I can replicate and investigate the issue

**Comment by rtivital** [comment]
Another suggestion will be to stick to older version of webpack (4.x), I've noticed that latest version of @rails/webpacker (5.4.0) uses webpack 5 which can be an issue.

**Comment by dsvanlani** [comment]
We've decided this is most likely an issue with our configuration.  I'm going to close the issue for now and we will follow up on this thread if we find a solution.

**Comment by rtivital** [comment]
Alright, feel free to reach out here of in Discord if you'll need any help

**Comment by lrhacker** [comment]
If anyone happens to find this thread and is in the same position, it wound up being an issue with our Babel configuration.

We had a large config file that was originally set up a few years ago, so we just started over with a new config and pulled in just what we needed to get our current app to build. There were a ton of unneeded plugins and settings based on the current state of our application.

We also ran into a second issue with Webpacker specifically, where it requires all Babel (and other build-related) packages to be in `dependencies`, not `devDependencies`, which we were able to track down an answer for [here](https://github.com/rails/webpacker/issues/2191#issuecomment-581574782).

We were still on the version of Webpacker that uses Webpack 4.x, so this might be different for anyone who is using the latest Webpacker with Webpack 5.x.
